test/NuGet.Core.Tests/NuGet.Build.Tasks.Test/GetRestoreProjectStyleTaskTests.cs

```diff
@@ -0,0 +1,173 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Linq;
+using FluentAssertions;
+using NuGet.Common;
+using NuGet.Configuration;
+using NuGet.ProjectModel;
+using NuGet.Test.Utility;
+using Xunit;
+
+namespace NuGet.Build.Tasks.Test
+{
+    public class GetRestoreProjectStyleTaskTests
+    {
+        [Fact]
+        public void Execute_WhenInvalidProjectStyleSupplied_LogsError()
+        {
+            var buildEngine = new TestBuildEngine();
+            var testLogger = buildEngine.TestLogger;
+
+            var task = new GetRestoreProjectStyleTask
+            {
+                BuildEngine = buildEngine,
+                RestoreProjectStyle = "Invalid"
+            };
+
+            task.Execute().Should().BeFalse();
+
+            testLogger.LogMessages.Should().Contain(i => i.Code == NuGetLogCode.NU1008 && i.Message == "Invalid project restore style 'Invalid'.");
```

Comment:
I don't see where this is raised: NU1008

Reply:
WarnAsError + this will cause some failure for some repos like corefx. 

https://github.com/dotnet/corefx/blob/master/eng/restore/repoRestore.targets#L4

I'll create a PR to fix it up on their end, I'm just worried about how many other potential issues we might have.

Reply:
Created a PR on their end https://github.com/dotnet/corefx/pull/42642.

Reply:
Actually I figured out that adding the error will cause problems in the Mono scenario since the error codes are in NuGet.Common.dll. So I decided to just treat an invalid value as Unknown for now to not break anyone who is setting it to something else. I just forgot to update the unit test after making this decision. That said, it’s still a good idea to update corefx.

Reply:
The mono scenario is messy regardless :( 
Having to avoid changes in shared libraries is sub-optimal at the minimum.